compiler/src/dotty/tools/dotc/core/Names.scala

```diff
@@ -293,6 +293,10 @@ object Names {
       else {
         if (Config.checkBackendNames) {
           if (!toStringOK) {
+            // We print the stacktrace instead of doing an assert directly,
+            // because asserts are caught in exception handlers which might
+            // cause other failures. In that case the first, important failure
+            // is lost.
             println("Backend should not call Name#toString, Name#mangledString should be used instead.")
```

Comment:
Use `Console.err.println` (or `System.err.println` instead). `printStackTrace` will print on stderr, and if you use println right before it which prints on stdout, the output might get mixed.